File: app/src/main/java/com/wipro/android/proficiencyexercise/domain/intractor/CanadaListUseCase.java
package com.wipro.android.proficiencyexercise.domain.intractor;

import com.wipro.android.proficiencyexercise.domain.repository.Repository;
import com.wipro.android.proficiencyexercise.model.CanadaList;
import javax.inject.Inject;

import io.reactivex.Observable;

public class CanadaListUseCase extends UseCase<CanadaList> {
    Repository repository;

    @Inject
    public CanadaListUseCase(Repository repository) {
        this.repository = repository;
    }

    @Override
    Observable<CanadaList> buildUseCaseObservable() {
        return repository.getCanadaList();
    }
}

File: app/src/main/java/com/wipro/android/proficiencyexercise/domain/intractor/CanadaListUseCase.kt
package com.wipro.android.proficiencyexercise.domain.intractor

import com.wipro.android.proficiencyexercise.data.remote.response.CanadaList
import com.wipro.android.proficiencyexercise.domain.repository.Repository
import io.reactivex.Observable
import javax.inject.Inject

class CanadaListUseCase @Inject
constructor(private val repository: Repository) : UseCase<CanadaList>() {

    override fun buildUseCaseObservable(): Observable<CanadaList> {
        return repository.canadaList
    }
}


